### PR TITLE
[CI] Build from OV wheel

### PR DESCRIPTION
**Pull request by mryzhov**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -105,6 +105,7 @@ python_abi = "none"
 [build-system]
 requires = [
     "py-build-cmake@git+https://github.com/tttapa/py-build-cmake@7ab73da351c7140f06d727a8705bece4cf544cd9",
-    "cmake~=3.15"
+    "cmake~=3.15",
+    "openvino~=2024.3.0.0.dev"
```
Comment (ilya-lavrenov) [inline pyproject.toml:109]: should we also update build instruction https://github.com/openvinotoolkit/openvino_tokenizers?tab=readme-ov-file#build-and-install-from-source to include that --extra-index-url is required?

**Review by ilya-lavrenov — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (mryzhov) [reply pyproject.toml:109]: added

File: README.md [hunk 2]
```diff
@@ -50,13 +50,12 @@ pip install --pre -U openvino openvino-tokenizers --extra-index-url https://stor
...
 ### Build and Install from Source
 
-Install [OpenVINO archive](https://docs.openvino.ai/2024/get-started/install-openvino.html) distribution. Use `--no-deps` to avoid OpenVINO installation from PyPI.
+openvino-tokenizers build depends on [openvino](https://pypi.org/project/openvino/) package which will be automatically installed from PyPI during the build process. To install unreleased versions, you would need to install openvino package from the nightly distribution channel using `--extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly`
 
 ```bash
-source path/to/installed/openvino/setupvars.sh
 git clone https://github.com/openvinotoolkit/openvino_tokenizers.git
 cd openvino_tokenizers
-pip install --no-deps .
```
Comment (apaniukov) [inline README.md:59]: We should have both methods here:
- install with openvino PyPI package
- install with openvino archive

File: README.md [hunk 3]
```diff
@@ -50,13 +50,12 @@ pip install --pre -U openvino openvino-tokenizers --extra-index-url https://stor
...
+pip install --no-deps . --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly
```
[the remark this answers is not in the document]
Comment (mryzhov) [reply README.md:58]: --no-deps skips installation dependencies only

[the remark this answers is not in the document]
Comment (apaniukov) [reply README.md:58]: Doesn't it break the archive installation scenario for several linux distros?

[the remark this answers is not in the document]
Comment (mryzhov) [reply README.md]: you are right, removed

File: README.md (hunk 2 above)
Comment (mryzhov) [reply README.md]: extended

**Review by apaniukov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@mryzhov 
can we do the same change for GenAI repo?

**Pull request merged**